lldb/include/lldb/Utility/RangeMap.h

```diff
@@ -493,36 +493,27 @@ class RangeDataVector {
 #ifdef ASSERT_RANGEMAP_ARE_SORTED
     assert(IsSorted());
 #endif
-    typename Collection::iterator pos;
-    typename Collection::iterator end;
-    typename Collection::iterator prev;
-    bool can_combine = false;
-    // First we determine if we can combine any of the Entry objects so we
-    // don't end up allocating and making a new collection for no reason
-    for (pos = m_entries.begin(), end = m_entries.end(), prev = end; pos != end;
-         prev = pos++) {
-      if (prev != end && prev->data == pos->data) {
-        can_combine = true;
-        break;
-      }
-    }
+    auto first_intersect = std::adjacent_find(
+        m_entries.begin(), m_entries.end(), [](const Entry &a, const Entry &b) {
+          return a.DoesAdjoinOrIntersect(b) && a.data == b.data;
```

Comment:
Is the equal check cheaper than the Intersect or Adjacency check? If so we should short circuit on equality before checking for an intersection

Reply:
Both of them are integer comparisons (in practice -- technically, this is a template so it could be whatever), so it really comes down to "which one is more likely to be false". I'm not sure about that, but I doubt this code is hot enough for it to matter. If we wanted to optimize this we could change the `DoesAdjoinOrIntersect` call to `b.GetRangeBase() <= a.GetRangeEnd()` (since the other check inside that function is guaranteed to be true due to sorting).